### PR TITLE
feat: Certificates Update

### LINKED ISSUE #289: Certificates update
### Feature Description

1. We updated the last 2 certificates to have `date` and `signature` placeholder.
2. We must also update the code on the backend of the certificate.
3. Add the new certificates created in the figma.
https://www.figma.com/file/jGpo62JSykdqBAKYQ8iiDg/Classroomio-Web?type=design&node-id=5421-26154&mode=design&t=1EwYGdsIKjHz1lxF-0

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by tunny17**
## What does this PR do?

Fixes #289 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- go to the /apps/backend dir then run "node server.js". Open your postman, set it up and input the params in the body and send the POST req.

### Required

- [ ] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/backend/src/utils/certificate.js [hunk 1]
```diff
@@ -112,7 +112,251 @@ function getPlainTheme({
...
+    <div class="pl-10 flex justify-between w-[70%] text-center my-[10%]">
+      <div>
+        <h1 class="h-2/4">DATE ISSUED:</h1>
+        <p class="border-b-2 border-black font-bold">02,May 2024</p>
+      </div>
+
+      <div>
+        <h1 class="h-2/4">FACILITATOR:</h1>
+        <p class="font-bold">ROTIMI BEST</p>
+      </div>
```
Comment (rotimi-best) [inline apps/backend/src/utils/certificate.js]: @tunny17 Lol, how will you add my name as the facilitator?

File: apps/backend/src/utils/certificate.js [hunk 2]
```diff
@@ -112,7 +112,251 @@ function getPlainTheme({
...
+      <h1>has succesfully completed training on</h1>
+      <h2 class="text-4xl leading-[2.5rem]">
+        ${courseName}
+      </h2>
+    </div>
+
+    <div class="pl-10 flex justify-between w-[70%] text-center my-[10%]">
+      <div>
+        <h1 class="h-2/4">DATE ISSUED:</h1>
+        <p class="border-b-2 border-black font-bold">02,May 2024</p>
```
Comment (rotimi-best) [inline apps/backend/src/utils/certificate.js]: Bro, and why did you hard code date in a dynamic certificate?

File: apps/backend/src/utils/certificate.js [hunk 3]
```diff
@@ -112,7 +112,251 @@ function getPlainTheme({
...
+      <div class="pl-10 flex justify-between w-[70%] text-center my-[10%]">
+        <div>
+          <h1 class="h-2/4">DATE ISSUED:</h1>
+          <p class="border-b-2 border-black font-bold">02,May 2024</p>
+        </div>
+
+        <div>
+          <h1 class="h-2/4">FACILITATOR:</h1>
+          <p class="font-bold">ROTIMI BEST</p>
+        </div>
```
Comment (rotimi-best) [inline apps/backend/src/utils/certificate.js]: Same issue with date and facilitator here.

just hide the facilitator section

File: apps/dashboard/src/lib/components/Org/Settings/index.svelte [hunk 4]
```diff
@@ -45,35 +45,35 @@
...
       {
         key: 0,
-        label: $t('settings.tabs.1'),
+        label: $t('settings.tabs.profile_tab'),
         tabKey: '',
         href: $page.url.pathname,
         disabled: false
       },
       {
         key: 1,
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Org/Settings/index.svelte]: This PR looks like a mix with some translation fixes.

Please pull the latest of main into this branch

**Review by rotimi-best — CHANGES_REQUESTED** [review]
Please update the branch with the latest of main and you need to fix the hard coded data

File: apps/dashboard/src/lib/components/Course/components/Ceritficate/certificates/PurpleProfessionalBadge.svelte [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<script>
+  import { course } from '$lib/components/Course/store';
+  import { currentOrg } from '$lib/utils/store/org';
+
+  export let studentName = '';
+  let logo = '/logo-512.png';
+
+  let dateIssued = new Date().toLocaleDateString();
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Ceritficate/certificates/PurpleProfessionalBadge.svelte]: These should not be `let` if it isn't changing

File: apps/dashboard/src/lib/components/Course/components/Ceritficate/certificates/BlueProfessionalBadge.svelte [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<script>
+  import { course } from '$lib/components/Course/store';
+  import { currentOrg } from '$lib/utils/store/org';
+
+  export let studentName = '';
+  let logo = '/logo-512.png';
+
+  let dateIssued = new Date().toLocaleDateString();
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Ceritficate/certificates/BlueProfessionalBadge.svelte]: `let` issue here, please also go to the other files to update.

File: apps/backend/static/logo.svg
Comment (rotimi-best) [inline apps/backend/static/logo.svg:1]: What's the purpose of this file? how is the logo shown in the other templates?

**Review by rotimi-best — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/lib/components/Course/components/Ceritficate/Design.svelte [hunk 7]
```diff
@@ -6,36 +6,71 @@
...
     isSaving = true;
-    if (!$course.description) {
-      errors.description = 'Description cannot be empty';
+
+    try {
+      courseSchema.parse({
+        description: $course.description || '',
+        is_certificate_downloadable: $course.is_certificate_downloadable || false,
+        certificate_theme: $course.certificate_theme || ''
+      });
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Ceritficate/Design.svelte]: @tunny17 Use the validation approach we use and not expose a schema to a component.

**Review by rotimi-best — CHANGES_REQUESTED** [review]

**Comment by rotimi-best** [comment]
@tunny17 
- [ ] Please take a look at all certificate with Date and please make this update.

Screenshot 2024-07-18 at 3 51 10 AM

- [ ] The logo and title name isn't aligned to the middle

image
image

**Comment by tunny17** [comment]
@rotimi-best  i've made the changes you requested

**Review by rotimi-best — APPROVED** [review]

File: apps/backend/src/utils/certificate.js [hunk 8]
```diff
@@ -99,21 +122,248 @@ function getPlainTheme({
...
+  courseName,
+  courseDescription,
+  orgLogoUrl,
+  orgName,
+  facilitator,
+  dateIssued
+}) {
+  const themeBody = `
+    <img
+      src="http://localhost:3002/purple-professional-badge-background.png"
```
Comment (rotimi-best) [inline apps/backend/src/utils/certificate.js]: Oh damin, I didn't notice this earlier.

You should NOT hard code a URL to localhost in your code that will be deployed to production 😢 

Please delete the static folder cause I will upload the images to cloudflare and provide you the URL.

Then delete this code:

```
// serve the images in the static folder
app.use(express.static('static'));
```

For this one use: https://assets.cdn.clsrio.com/certificates/purple-professional-badge-background.png

Then replace all the templates you have used localhost.

**Review by rotimi-best — CHANGES_REQUESTED** [review]

**Review by rotimi-best — APPROVED** [review]

**Pull request merged**